### PR TITLE
M276 Combat Toolbelt Rig

### PR DESCRIPTION
**Pull request by Sulaboy**
Adds the M276 pattern combat toolbelt rig as to the game as a new belt for combat engineers and some other roles.

The M276 pattern combat toolbelt is an alternate version of the M276 pattern toolbelt with some combat capabilities. This rig is capable of carrying fives tool and has a holster for a handgun. 

# Explain why it's good for the game

Provides engineers with more choices for tool storage.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

combatbelt showcase

The newest iteration of this belt has total slots, and cannot fit pistol magazines.
</details>

**Disclaimer, I barely know how to code. If I need to make any tweaks please be patient with me.**

# Changelog

:cl:
add: Added item M276 pattern combat toolbelt rig as a point buy item (15 points) to combat engineer vendor.
imageadd: Sprites for M276 pattern combat toolbelt rig
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/pistols.dm [hunk 1]
```diff
@@ -47,7 +47,7 @@
 
 /obj/item/weapon/gun/pistol/m4a3
 	name = "\improper M4A3 service pistol"//1911
-	desc = "An M4A3 Service Pistol, once the standard issue sidearm of the Colonial Marines but has recently been replaced with the 88 Mod 4 combat pistol. Fires 9mm pistol rounds."
+	desc = "An M4A3 Service Pistol, a stinky test pistol for testing changing code with a running server and being stinky."
```
Comment (Zonespace27) [inline code/modules/projectiles/guns/pistols.dm:50]: .

File: code/modules/projectiles/guns/rifles.dm [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
 /obj/item/weapon/gun/rifle/m41a
 	name = "\improper M41A pulse rifle MK2"
-	desc = "The standard issue rifle of the Colonial Marines. Commonly carried by most combat personnel. Uses 10x24mm caseless ammunition."
+	desc = "Stinky Gun Test Gun Test Stinky. Commonly carried by most combat personnel. Uses 10x24mm caseless ammunition."
```
Comment (Zonespace27) [inline code/modules/projectiles/guns/rifles.dm:40]: .

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/pistols.dm (hunk 1 above)
Comment (Sulaboy) [reply code/modules/projectiles/guns/pistols.dm:50]: That wasn't supposed to be there.

File: code/modules/projectiles/guns/rifles.dm (hunk 2 above)
Comment (Sulaboy) [reply code/modules/projectiles/guns/rifles.dm]: Fixed

**Review by morrowwolf — CHANGES_REQUESTED** [review]
GA review:

This is far too powerful and power creep in its current state. This belt needs to be balanced alongside other belts rather than be a straight upgrade.

**Comment by Sulaboy** [comment]
> GA review:
> 
> This is far too powerful and power creep in its current state. This belt needs to be balanced alongside other belts rather 
than be a straight upgrade.

Didn't get this to you before you got off for the night. Here is the picture with the belt capacity comparisons. I'll definitely be cutting out a magazine and slot from the belt at least or both if you think that's the way to go. Let me know if you would like it to be a point buy as well. I'll make any changes to this on Friday.

asdf

**Comment by Sulaboy** [comment]
Removed a slot from the belt to put it more in line with other belts, and added a point cost. Removed any addition of the belt from other vendors and WO spawns.

**Review by morrowwolf — COMMENTED** [review]
GA Approved. Usual disclaimer: maintainer team can and will change this item in whatever way (including completely reverted) if it effects balance in a way unforeseen.

**Comment by NewyearnewmeUwu** [comment]
IMO, this should be the lifesaver bag to the medical rig or vice versa. A sidegrade, not an upgrade or a downgrade. There are some things the medical rig legitimately does better, but most of the time the livesaver is the nobrainer pick.

**Review by Agameofscones — CHANGES_REQUESTED** [review]
should be `+combatutility_half`. Current is incorrect as `+combatutility`. (sprite state names)

Otherwise sprites checkout. Ping me when you've fixed it and I'll approve.

File: code/game/objects/items/storage/belt.dm [hunk 3]
```diff
@@ -1546,6 +1546,50 @@
 	bypass_w_limit = list(/obj/item/mortar_shell)
 	has_gamemode_skin = TRUE
 
+/obj/item/storage/belt/gun/utility
+	name = "\improper M276 pattern combat toolbelt rig"
+	desc = "The M276 pattern combat toolbelt rig is an alternative load-bearing equipment of the USCM for engineers conducting repairs within combat zones. It consists of a modular belt with various clips and pouches for tools along with a holster for a sidearm. Due to the bulk of the sidearm, it is unable to hold as many tools as its standard counterpart."
+	storage_slots = 6
+	icon_state = "combatutility"
```
Comment (Zonespace27) [inline code/game/objects/items/storage/belt.dm:1553]:
```suggestion
	icon_state = "combatutility"
```

**Review by Zonespace27 — APPROVED** [review]

**Comment by Zonespace27** [comment]
Looks good

**Pull request merged**